### PR TITLE
fix reading profiles bug

### LINKED ISSUE #351: Inspec doesn't read controls
What used to be 

``` interactive
$ inspec check base_linux 
I, [2015-12-31T12:55:37.496650 #86922]  INFO -- : Checking profile in base_linux
W, [2015-12-31T12:55:37.496768 #86922]  WARN -- : The use of `metadata.rb` is deprecated. Use `inspec.yml`.
I, [2015-12-31T12:55:37.496801 #86922]  INFO -- : Metadata OK.
D, [2015-12-31T12:55:37.496887 #86922] DEBUG -- : Found 45 rules.
D, [2015-12-31T12:55:37.496904 #86922] DEBUG -- : Verify all rules in  base_linux/tests/filesystem_spec.rb
D, [2015-12-31T12:55:37.496923 #86922] DEBUG -- : Verify all rules in  base_linux/tests/sysctl_ipv4_spec.rb
D, [2015-12-31T12:55:37.496956 #86922] DEBUG -- : Verify all rules in  base_linux/tests/sysctl_ipv6_spec.rb
D, [2015-12-31T12:55:37.496974 #86922] DEBUG -- : Verify all rules in  base_linux/tests/sysctl_spec.rb
```

is now

``` interactive
$ inspec check base_linux
I, [2016-01-07T13:10:11.392420 #14414]  INFO -- : Checking profile in base_linux/
W, [2016-01-07T13:10:11.392544 #14414]  WARN -- : Missing profile maintainer in inspec.yml
W, [2016-01-07T13:10:11.392592 #14414]  WARN -- : No controls or tests were defined.
```

This is wrong.

### PR DESCRIPTION
**Pull request by srenatus**
For reading the profiles metadata, we're using the train mock backend
through Inspec::Runner. The new `supports` feature never agrees with the
mock backend.

Now, it we figure out if this is a mock class and then just say that it
supports whatever we're asking for.

Tl;dr: there's probably a more beautiful solution to this.

Added a test case, but it fails -- while the command line interface
works fine.

Fixes #351.

### CODE REVIEW

**Comment by srenatus** [comment]
Ok, this breaks many of our other tests... because we're setting up a "mock-ish" ubuntu backend (for example), that is mock enough to trigger the `return true` of the supports check.

**Comment by chris-rock** [comment]
I extended the PR to make the support check optional.

**Comment by chris-rock** [comment]
this fix leaves #354

**Comment by arlimus** [comment]
Looks great, thank you both for the fix! :+1: :)
